docs/adoption.md

```diff
@@ -17,39 +19,243 @@ Clone/fork the following repositories:
   - Use a region which is geographically close to the users/residents and the 3rd party data
 - Azure app service for each API
   - Create a 'staging' deployment slot
   - Each app service (and it's deployment slot) will have a publish profile which needs to be added to the repository secrets[^1]
 - Cosmos DB
-    - 2 databases - production and staging
-    - Each with `repair-requests` collections
+  - 2 databases - production and staging
+  - Each with `repair-requests` collections
 - Blob storage account
-    - Create 2 containers - production and staging
+  - Create 2 containers - production and staging
 
 ### Frontend
+
 - Create Azure Static Web App
   - A deployment token will need to be added to the repository secrets[^1]
 
+## Deploying to Azure
+
+An Azure service principal is an identity created that can be used for automated tools such as GitHub actions to access Azure resources. This access is restricted by the roles assigned to the service principal, giving you control over which resources can be accessed and at which level. In order to assign access to the service principal, you must have the necessary access to so.
+
+### Create a service principal
+
+1. Ask your subscription administrator to add you to the User Access Administrator role. [See documentation](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal#check-azure-subscription-permissions). (_If the subscription administrator is creating the service principal, ensure that you request the environment variables as specified in the Adding GitHub actions job and secrets step_)
+2. Create the service principal. [See documentation](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal#register-an-application-with-azure-ad-and-create-a-service-principal).
+3. Assign a role to the service principal as per [this documentation](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal#assign-a-role-to-the-application). If you are not able to do this, contact your subscription administrator as you do not have the permission to do so
+4. Create a client secret for your service principal. [See documentation](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal#option-2-create-a-new-application-secret). Make sure to keep a note of the secret or copy it to your clipboard
+
+### Create a terraform backend
+
+1. Create a resource group for your infrastructure or use an existing one where your resources can be deployed
+2. Create a storage account, and attach it to the resource group you created above
+3. In that storage account, create a container called tfstate. This is where your state file will live
+4. Create a terraform directory followed by a main.tf file. Add the following terraform (Note, the terraform version may vary):
```

Comment:
Given [main.tf](https://github.com/Newarkandsherwood/housing-repairs-online-frontend/blob/terraform-ci/terraform/main.tf) has already been created, this and the following step are not required.